### ISSUE TITLE
Unity app crashes after 30s / when sending a heratbeat

### ISSUE THREAD

**Issue by namxam** [Mazyod/PhoenixSharp#29]
I am a complete Unity and C# newbie, but I managed to create an app, install your plugin and establish a connection. But it seems that I did something stupid, as the app is always crashing when a heartbeat is sent. It took me a while to find the culprit, but when I set `socketOptions.HeartbeatInterval = null;` it works flawlessly. When changing its value or keeping its default I can reproduce those crashes. Any idea what may cause it?

I guess it has to do with where I create the connection. Am I blocking the main thread? Where would I create a connection?

Well, I just discovered, that it is also crashing when joining a channel. I am completly lost. Anyone with an idea?

**Comment by namxam** [comment]
Well, it seems to be an issue with the dot net websocket adapter. After switching to the WebSocketSharp implementation, everything seems to be working.

**Comment by Mazyod** [comment]
Thanks for using the library and sharing your experience, @namxam. I'll add a disclaimer regarding the instability of the DotNetAdapter, I also believe it's unstable (at least for .netstandard2.0). The main reason we have it is the promise of a dependency free implementation, but in practice, I have found it hard to deal with.

Nothing wrong with using WebsocketSharp if it's working for you, but I remember facing issues with callbacks running on a background thread when I tried it.

Just in case you missed it, I personally use BestHTTP with my Unity game and would recommend it from experience. I have also seen other promising projects such as [Mirror][mirror-lib] (I haven't tried it yet, tho).

[mirror-lib]: https://github.com/MirrorNetworking/Mirror
